src/utils/useFieldState.ts

```diff
@@ -0,0 +1,50 @@
+import React, {useCallback, useState} from "react";
+import {SkjemaelementFeil} from "nav-frontend-skjema/lib/skjemaelement-feilmelding";
+
+export interface FieldState {
+    input: {
+        value: string;
+        onChange: React.ChangeEventHandler;
+        onBlur: React.FocusEventHandler;
+        feil?: SkjemaelementFeil;
+    };
+    validate(): void;
+}
+
+export default function useFieldState(initialState: string, validate: (value: string) => string | null = () => null): FieldState {
```

Comment:
Kan vi forenkle signaturen ? 

kanskje vi kan ta ut validate typen i en egen `type` så er det litt enklere å forstå?

`() => null` er jo en Noop func ? Kanskje vi kan lage denne utils funksjonen og bruke det her? Tenker det kanskje blir lettere å lese

Reply:
Alà `useFieldState(initialState: string, validate: Validator = NoopValidator): FieldState` ??

Reply:
erstattet "null = () => null" med "nullop" etter å ha definert "export type nullop = () => null ;" Det gir følgende feilmelding: 
TypeScript error in C:/Users/Y130311/projects/fo-dialog/src/view/DialogNew.tsx(20,36):
Argument of type '(value: string) => "Tema må ha innhold." | null' is not assignable to parameter of type '(value: string) => string | nullop'.
  Type '"Tema må ha innhold." | null' is not assignable to type 'string | nullop'.
    Type 'null' is not assignable to type 'string | nullop'.  TS2345

tilsvarende med stringop

Reply:
Erstattet første linjen i useFieldState med disse 3:

export type Validator = (value: string) => string | null;
const noopValidator: Validator = () => null;
export default function useFieldState(initialState: string, validate: Validator = noopValidator): FieldState {